test/Dialect/FIRRTL/canonicalization.mlir

```diff
@@ -963,4 +963,188 @@ firrtl.module @ComparisonOfUnsized(
+  // CHECK-NEXT: firrtl.connect %y5, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y6, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y7, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y8, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y9, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y10, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y11, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y12, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y13, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y14, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y15, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y16, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y17, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y18, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y19, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y20, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y21, %c1_ui1
+  // CHECK-NEXT: firrtl.connect %y22, %c0_ui1
+  // CHECK-NEXT: firrtl.connect %y23, %c1_ui1
+}
+
+// CHECK-LABEL: @ComparisonOfZeroWidths
+firrtl.module @ComparisonOfZeroWidths(
+  %xu0: !firrtl.uint<0>,
+  %xu1: !firrtl.uint<0>,
+  %xs0: !firrtl.sint<0>,
+  %xs1: !firrtl.sint<0>,
+  %y0: !firrtl.flip<uint<1>>,
+  %y1: !firrtl.flip<uint<1>>,
+  %y2: !firrtl.flip<uint<1>>,
+  %y3: !firrtl.flip<uint<1>>,
+  %y4: !firrtl.flip<uint<1>>,
+  %y5: !firrtl.flip<uint<1>>,
+  %y6: !firrtl.flip<uint<1>>,
+  %y7: !firrtl.flip<uint<1>>,
+  %y8: !firrtl.flip<uint<1>>,
+  %y9: !firrtl.flip<uint<1>>,
+  %y10: !firrtl.flip<uint<1>>,
+  %y11: !firrtl.flip<uint<1>>,
+  %y12: !firrtl.flip<uint<1>>,
+  %y13: !firrtl.flip<uint<1>>,
+  %y14: !firrtl.flip<uint<1>>,
+  %y15: !firrtl.flip<uint<1>>,
+  %y16: !firrtl.flip<uint<1>>,
+  %y17: !firrtl.flip<uint<1>>,
+  %y18: !firrtl.flip<uint<1>>,
+  %y19: !firrtl.flip<uint<1>>,
+  %y20: !firrtl.flip<uint<1>>,
+  %y21: !firrtl.flip<uint<1>>,
+  %y22: !firrtl.flip<uint<1>>,
+  %y23: !firrtl.flip<uint<1>>
+) {
+  // CHECK-NEXT: [[_:.+]] = firrtl.constant
+  // CHECK-NEXT: [[_:.+]] = firrtl.constant
+
+  %0 = firrtl.leq %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %1 = firrtl.leq %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+  %2 = firrtl.lt %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %3 = firrtl.lt %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+  %4 = firrtl.geq %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %5 = firrtl.geq %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+  %6 = firrtl.gt %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %7 = firrtl.gt %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+  %8 = firrtl.eq %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %9 = firrtl.eq %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+  %10 = firrtl.neq %xu0, %xu1 : (!firrtl.uint<0>, !firrtl.uint<0>) -> !firrtl.uint<1>
+  %11 = firrtl.neq %xs0, %xs1 : (!firrtl.sint<0>, !firrtl.sint<0>) -> !firrtl.uint<1>
+
```

Comment:
There could make smaller by anding all the results together for a single output.

Reply:
That would yield a single checksum value, and if a change introduced multiple mistakes that happened to sum up to the same number of folds-to-1 and folds-to-0, we'd miss that.